File: app/src/main/java/com/live/myapplication/LiveDataViewModel.java
package com.live.myapplication;

import android.os.SystemClock;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.Date;
import java.util.Timer;
import java.util.TimerTask;

public class LiveDataViewModel extends ViewModel {
    private MutableLiveData<Long> mDelayTime = new MutableLiveData<Long>();

    private final Timer timer;

    public LiveDataViewModel(long milli) {
        timer = new Timer();

        // Update the elapsed time every second.
        timer.schedule(new TimerTask() {//scheduleAtFixedRate
            @Override
            public void run() {
                Long cur = System.currentTimeMillis();
                mDelayTime.postValue(cur);
            }
        }, 0, milli);


    }

    public LiveData<Long> getDelayTime() {
        return mDelayTime;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        timer.cancel();
    }
}

File: app/src/main/java/com/live/myapplication/LiveDataViewModel.kt
package com.live.myapplication

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import java.util.*

class LiveDataViewModel(milli: Long) : ViewModel() {
    private val mDelayTime = MutableLiveData<Long>()
    private val timer: Timer
    val delayTime: LiveData<Long>
        get() = mDelayTime

    public override fun onCleared() {
        super.onCleared()
        timer.cancel()
    }

    init {
        timer = Timer()

        // Update the elapsed time every second.
        timer.schedule(object : TimerTask() {
            //scheduleAtFixedRate
            override fun run() {
                val cur = System.currentTimeMillis()
                mDelayTime.postValue(cur)
            }
        }, 0, milli)
    }
}